src/Components/test/testassets/BasicTestApp/AuthTest/ServerAuthenticationStateProvider.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+
+namespace BasicTestApp.AuthTest
+{
+    // This is intended to be similar to the authentication stateprovider included by default
+    // with the client-side Blazor "Hosted in ASP.NET Core" template
```

Comment:
Now I've written out this code, it makes me think we should consider making this more built-in, because it doesn't feel very friendly right now. We could ship a standard `ServerAuthenticationStateProvider` which developers explicitly add to their client-side Blazor DI config, along with some option about what URL it hits to get the state data. We'd also ship a standard `ClientSideAuthenticationStateData` DTO type (probably with a better name). Then the amount of boilerplate code in apps/templates would be pretty minimal (just some DI config plus `UserController` on the server), plus it's still possible to make your own entirely custom `AuthenticationStateProvider` if you don't like ours. Thoughts on this?